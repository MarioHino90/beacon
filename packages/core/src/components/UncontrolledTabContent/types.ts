import { TabContentProps } from "../TabContent";
import { TabProps } from "../Tab";

export interface UncontrolledTabContentProps
  extends Omit<TabContentProps, "active"> {
  /** the id of the current tab */
  tabId: TabProps["tabId"];
}
