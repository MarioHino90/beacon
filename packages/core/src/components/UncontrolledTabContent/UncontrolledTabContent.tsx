/**
 * @fileoverview an uncontrolled version of `TabContentProps`
 */
import React, { useContext, ReactElement } from "react";

import { TabContent } from "../TabContent";
import { UncontrolledTabsContext } from "../UncontrolledTabsContext";

import { UncontrolledTabContentProps } from "./types";

export const UncontrolledTabContent = ({
  tabId,
  ...props
}: UncontrolledTabContentProps): ReactElement => {
  const { activeTab } = useContext(UncontrolledTabsContext);

  return <TabContent active={activeTab === tabId} {...props} />;
};
