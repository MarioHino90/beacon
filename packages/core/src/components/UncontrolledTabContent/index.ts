export { UncontrolledTabContent } from "./UncontrolledTabContent";

export type { UncontrolledTabContentProps } from "./types";
