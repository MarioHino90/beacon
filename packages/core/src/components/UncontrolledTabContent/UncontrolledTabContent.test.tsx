import React from "react";
import { shallow } from "enzyme";
import { BodyText } from "@braze/beacon-core";

import { UncontrolledTabContent } from "./UncontrolledTabContent";

jest.mock("react", () => {
  const react = jest.requireActual("react");

  return {
    ...react,
    useContext: jest.fn().mockReturnValue({
      activeTab: 2,
      onTabChange: jest.fn().mockName("onTabChange"),
    }),
  };
});

describe("UncontrolledTabContent", () => {
  afterEach(jest.clearAllMocks);

  it("should render correctly", () => {
    const wrapper = shallow(
      <UncontrolledTabContent tabId={1}>
        <BodyText>Hello World!</BodyText>
      </UncontrolledTabContent>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should render correctly with className", () => {
    const wrapper = shallow(
      <UncontrolledTabContent tabId={1} className="extraClass">
        <BodyText>Hello World!</BodyText>
      </UncontrolledTabContent>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should render correctly if active", () => {
    const wrapper = shallow(
      <UncontrolledTabContent tabId={2}>
        <BodyText>Hello World!</BodyText>
      </UncontrolledTabContent>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
