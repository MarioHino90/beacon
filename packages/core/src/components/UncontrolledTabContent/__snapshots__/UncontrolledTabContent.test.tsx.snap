// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UncontrolledTabContent should render correctly 1`] = `
<TabContent
  active={false}
>
  <BodyText>
    Hello World!
  </BodyText>
</TabContent>
`;

exports[`UncontrolledTabContent should render correctly if active 1`] = `
<TabContent
  active={true}
>
  <BodyText>
    Hello World!
  </BodyText>
</TabContent>
`;

exports[`UncontrolledTabContent should render correctly with className 1`] = `
<TabContent
  active={false}
  className="extraClass"
>
  <BodyText>
    Hello World!
  </BodyText>
</TabContent>
`;
